### ISSUE TITLE
Conflicting variable names in taco tool pseudo-code output

### ISSUE THREAD

**Issue by stephenchouca** [tensor-compiler/taco#78]
For instance, for the addition of two DCSR matrices:

```
taco "A(i,j) = B(i,j) + C(i,j)" -f=A:ss -f=B:ss -f=C:ss
```

taco tool produces the following pseudo-code:

```c++
int A1_pos = A0_pos[0];
int A2_pos = A1_pos[A1_pos];  // A1_pos has multiple meanings here
int B1_pos = B0_pos[0];
int C1_pos = C0_pos[0];
while ((B1_pos < B0_pos[1]) && (C1_pos < C0_pos[1])) {
  int iB = B0_idx[B1_pos];
  int iC = C0_idx[C1_pos];
  int i = min(iB, iC);
  if ((iB == i) && (iC == i)) {
    int B2_pos = B1_pos[B1_pos];
    int C2_pos = C1_pos[C1_pos];
...
```

**Comment by shoaibkamil** [comment]
Can you see what the output looks like at commit 2a118bc?  This may be related to recent IR changes.

**Comment by stephenchouca** [comment]
The issue appears at that commit as well.

**Comment by fredrikbk** [comment]
This should be fixed in cc6932f9524173bebb0d21f8534d2284d7f6b639
